monkey/infection_monkey/island_api_client/http_island_api_client.py

```diff
@@ -45,16 +45,33 @@ class HTTPIslandAPIClient(IIslandAPIClient):
     """
 
     def __init__(
-        self, agent_event_serializer_registry: AgentEventSerializerRegistry, http_client: HTTPClient
+        self,
+        agent_event_serializer_registry: AgentEventSerializerRegistry,
+        http_client: HTTPClient,
+        otp: str,
     ):
         self._agent_event_serializer_registry = agent_event_serializer_registry
         self.http_client = http_client
+        self._otp = otp
```

Comment:
Do we have to keep the OTP? Why don't we just read the environment variable in `get_authentication_token()` and make the request to get the auth token?

Reply:
It's not the responsibility of `IIslandAPIClient` to read and validate environmental variables. Reading and validating cmd parameters and environmental variables should be a responsibility of a different component and that should be done at construction time, not during a runtime. This way if we change where we get an OTP from (maybe we pass it as a cmd or push it to a file) we don't need to change this code and vice-versa, if we change the client to communicate via ssh for example, we don't need to copy paste the code to extract the OTP.

Reply:
@shreyamalviya I don't follow you. The OTP is injected into the `IslandAPIClient`. There's no need for a separate component that deals with OTPs to be passed here.

Reply:
You're right. What I don't like here is that we're storing the OTP when it's only relevant and useful for a few minutes.

Reply:
Yes, the OTP is only used in `connect`, but a different `IIslandAPIClient` might use a different authentication mechanism, so we can't pass it into `connect`. We could clear the OTP once it's used, but I'm not sure what we gain from it